cli/2024-10-cli-udt.md

```diff
@@ -0,0 +1,223 @@
+
+1. Create a resource provider from a file 
+    
+    ```bash
+    rad resource-provider create -f Mycompany.Messaging.yaml
+    Resource provider Mycompany.Messaging created successfully
+    ```
+
+2. Create a resource provider with flags
+
+    ```bash
+    rad resource-provider create --name Mycompany.Messaging 
+    Resource provider Mycompany.Messaging created successfully
+    ```
+
+**Flags**
+- `-f`, `--from-file` : File path to the schema definition file
+- `-n`, `--name` : Name of the resource provider
+
+Question:
+- Should there be a flag that indicates the tenant to which the resource provider is created? Might be needed for multi
+
+#### 2. rad resource-provider list
+Resource providers are entities that implement a group of resource types such as `Applications.Core`. Users can list all the resource providers available in your tenant.
+
+    ```bash
+    rad resource-provider list 
+    RESOURCE-PROVIDER   TYPE                                STATE
+    Mycompany.Messaging System.Resources/resourceProviders  Succeeded
+    ```
+
+#### 3. rad resource-provider show
+Resource providers are entities that implement a group of resource types such as `Applications.Core`. User can display the details of the resource providers.
+
+    ```bash
+    rad resource-provider show Mycompany.Messaging
+    RESOURCE-PROVIDER   TYPE                                STATE
+    Mycompany.Messaging System.Resources/resourceProviders  Succeeded
+    ```
+
+#### 4. rad resource-provider delete
+Resource providers are entities that implement a group of resource types such as `Applications.Core`. Users can delete a resource provider.
+
+    ```bash
+    rad resource-provider delete Mycompany.Messaging
+    ```
+**Behaviors**
+Option 1 : Radius can prompt the user with a warning message if there are resources provisioned and running under the resource provider. If the user confirms deletion, Radius cleans up everything, deletes resources, types and resource providers.
+
+Option 2 : Radius can error out if there are resources provisioned and running under the resource provider. The user needs to delete the resources before deleting the resource provider.
+
+### rad resource-type
+
+#### 1. rad resource-type create
+Resource types are the entities that implement resource types such as 'Applications.Core/containers'. Each resource type can define multiple API versions, and each API version defines a schema that resource instances conform to. Inputs can be passed in as a file or as flags.
+
+1. Create a resource type from a file 
+    
+    ```bash
+    rad resource-type create -f plaid.yaml
+    ```
+
+2. Create a resource type from flags
+
+    ```bash
+    rad resource-type create --resource-provider Mycompany.Messaging --resource-type plaidResource --version=2024-10-09 --properties host=string port=string
```

Comment:
I'm skeptical of this idea because I don't want to build two authoring experiences. It's going to be very hard to replicate the power of the written YAML manifest.

Is is **absolutely critical** to have?

Reply:
Agree that this is not a scalable solution. We could do interactive experiences in the future based on user feedback and adoption